cico_pr_build.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+REGISTRY_URI="quay.io"
+REGISTRY_NS="fabric8io"
+REGISTRY_IMAGE="fabric8-online-docs"
+BUILDER_IMAGE="docs-builder"
+
+if [ "$TARGET" = "rhel" ]; then
+    REGISTRY_URL=${REGISTRY_URI}/openshiftio/rhel-${REGISTRY_NS}-${REGISTRY_IMAGE}
+    DOCKERFILE_DEPLOY="Dockerfile.deploy.rhel"
+else
+    REGISTRY_URL=${REGISTRY_URI}/openshiftio/${REGISTRY_NS}-${REGISTRY_IMAGE}
+    DOCKERFILE_DEPLOY="Dockerfile.deploy"
+fi
+
+TARGET_DIR="html"
+
+function docker_login() {
+    local USERNAME=$1
+    local PASSWORD=$2
+    local REGISTRY=$3
+
+    if [ -n "${USERNAME}" ] && [ -n "${PASSWORD}" ]; then
+        docker login -u ${USERNAME} -p ${PASSWORD} ${REGISTRY}
+    fi
+}
+
+function tag_push() {
+    local TARGET_IMAGE=$1
+
+    docker tag ${BUILDER_IMAGE} ${TARGET_IMAGE}
+    docker push ${TARGET_IMAGE}
+}
+
+function addCommentToPullRequest() {
+    message="$1"
+    pr="$2"
+    project="$3"
+    url="https://api.github.com/repos/${project}/issues/${pr}/comments"
+
+    set +x
+    echo curl -X POST -s -L -H "Authorization: XXXX|base64 --decode)" ${url} -d "{\"body\": \"${message}\"}"
+    curl -X POST -s -L -H "Authorization: token $(echo ${FABRIC8_HUB_TOKEN}|base64 --decode)" ${url} -d "{\"body\": \"${message}\"}"
+    set -x
+}
+
+# Exit on error
+set -ex
+
+eval "$(./env-toolkit load -f jenkins-env.json \
+        FABRIC8_HUB_TOKEN \
+        QUAY_USERNAME \
+        QUAY_PASSWORD \
+        ghprbGhRepository \
+        ghprbPullAuthorLogin \
+        ghprbPullId \
+        BUILD_ID)"
+
+# We need to disable selinux for now, XXX
+/usr/sbin/setenforce 0 || :
+
+# Get all the deps in
+yum -y install docker make git
+service docker start
+
+#REMOVE DIR WITH OLD BUILT DOCS IF IT EXISTS
+if [ -d $TARGET_DIR ]; then rm -rf ${TARGET_DIR}/; fi
+
+#CREATE A FRESH DIR FOR BUILT DOCS TO BE PUT INTO
+mkdir -pm 777 ${TARGET_DIR}
+
+#BUILD BUILDER IMAGE
+docker build -t ${BUILDER_IMAGE} -f Dockerfile.build .
+
+#LOGIN IS REQUIRED FOR BUILDING ON RHEL
+docker_login "${QUAY_USERNAME}" "${QUAY_PASSWORD}" "${REGISTRY_URI}"
+
+#PUSH
+TAG= "SNAPSHOT-PR-${ghprbPullId}-${BUILD_ID}"
+tag_push "${REGISTRY_URL}:${TAG}"
+
```

Comment:
how do you plan to address the preview part ?

Reply:
I think we will get an account of dev cluster for this, configure that in CICO (vault), and will deploy the image there using openshift-template, and comment the route.

Reply:
>how do you plan to address the preview part ?

@chmouel https://github.com/redhat-developer/rh-che/pull/619/files  is a very complicated example but we can do something similar to it.

Reply:
Nice one!, make sure you generate a template with a dynamic route that has the PR id so they don't overwrite! (and post back the right link into the say issue)

Reply:
I think creating a new namespace for every PR is perfect way, and delete the namespace whenver PR is merged or closed otherwise that will remain running forever